scripts/xtensa-build-zephyr.py

```diff
@@ -323,6 +324,14 @@ def west_init_update():
 	# Do NOT "west update sof"!!
 	execute_command(["west", "update", "zephyr", "hal_xtensa"], timeout=300, cwd=west_top)
 
+def get_sof_version():
+	"""[summary] Extract SOF firmware file version string (x.y or x.y.z) from latst git tag
+	by 'git describe --tags' and regex match
+	"""
+	sof_tag = subprocess.check_output(["git", "describe", "--tags"]).strip().decode()
+	sof_version = re.search('([0-9]+)[.]([0-9]+)([.]([0-9]+))?', sof_tag)
```

Comment:
Please don't copy/paste/diverge the code already in `scripts/cmake/version.cmake`. It's more complex than this and already different from this. Also, `git describe` can take time.

Instead parse the output of `scripts/cmake/version.cmake` like this (tested):
```python
defines = {}
with open("zephyrproject/build-$platform/zephyr/include/generated/sof_versions.h") as hfile:
    for hline in hfile:
        words = hline.split()
        if words[0] == '#define':
            defines[words[1]] = words[2]

print(defines)

{'SOF_MAJOR': '3', 'SOF_MINOR': '2', 'SOF_MICRO': '1', 'SOF_TAG': ...
```

Reply:
@marc-hb I updated the PR to get version from sof_version.h as you suggested, and we can always pass a 3-field version string to rimage.
@aiChaoSONG @lgirdwood We no longer need the rimage PR https://github.com/thesofproject/rimage/pull/95 so I closed it.